interview-questions.md

```diff
@@ -8,44 +8,60 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What does calling super do in a Ruby class?
 
-Your answer:
+Your answer: Calling super copies the paramaters from the previous class that you initiated. 
 
-Researched answer:
+Researched answer: In ruby, calling super invokes the same method defined in the superclass, it allows the sublcass to inherit the behavior of the superclass method. 
 
-2. What is a gem?
+2. What is a gem? 
 
-Your answer:
+Your answer; Its a library of customs codes you can use. Kind of like yarn but for ruby. 
 
-Researched answer:
+Researched answer: A "gem" refers to a library that can be installed and used in ruby applications, gems are reusable code someone else wrote. 
 
 3. What is a relational database? Are there other kinds of databases?
 
-Your answer:
+Your answer: A relational datatbase is a database that is 
+(honestly I cant think of an answer just off the top of my head)
+Im thinkig sql. 
 
-Researched answer:
+Researched answer: A database modell that stores and organizes data. Its structured into tables with rows and columns. Some examples of databases:
+NoSQL: this is a database that does not use the traditional model. 
+Columnar databases: This uses columns rather than rows
+Document-oriented databases: These store data in a document format such as JSON. These are useful for storing data that doesnt fit in columns or rows.
 
 4. What are primary keys? Why are they important?
 
-Your answer:
 
-Researched answer:
+Your answer: Same with this one Just off the top of my head I dont really know. 
+
+Researched answer: A primary key is a field in a database that makes every record unique. It serves as an identifier for each row and gives us a way to retrieve, update, and delete data in the database. 
+Example: "id" could be used as increasing unique integer value in a table.
+In our challenges from last week, during our credit card challange we used the Id primary key to assign each credit card to an owner. 
+
```

Comment:
I like the personal touch by adding in your personal experience using the primary key, nice job!